src/log-store/src/kafka/topic_client_manager.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use common_config::wal::kafka::{KafkaOptions, KafkaTopic as Topic};
+use dashmap::mapref::entry::Entry as DashMapEntry;
+use dashmap::DashMap;
+use rskafka::client::partition::{PartitionClient, UnknownTopicHandling};
+use rskafka::client::producer::aggregator::RecordAggregator;
+use rskafka::client::producer::{BatchProducer, BatchProducerBuilder};
+use rskafka::client::{Client, ClientBuilder};
+use rskafka::BackoffConfig;
+use snafu::ResultExt;
+
+use crate::error::{BuildKafkaClientSnafu, BuildKafkaPartitionClientSnafu, Result};
+
+// There's only one partition for each topic currently.
+const DEFAULT_PARTITION: i32 = 0;
+
+pub type TopicClientRef = Arc<TopicClient>;
+pub type TopicClientManagerRef = Arc<TopicClientManager>;
+
+#[derive(Debug)]
+pub struct TopicClient {
+    /// The raw client used to construct a batch producer and/or a stream consumer for a specific topic.
+    pub raw_client: Arc<PartitionClient>,
+    /// A producer used to buffer log entries for a specific topic.
+    pub producer: Arc<BatchProducer<RecordAggregator>>,
+}
```

Comment:
A `PartitionClient` is responsible for handling accesses to a specific partition of a topic. Since a topic only manages one partition by design, the name `TopicClient` seems good to me. You may propose a better name.